session.go

```diff
@@ -722,6 +794,56 @@ func (s *session) handleUnpackedPacket(packet *unpackedPacket, rcvTime time.Time
 
 	r := bytes.NewReader(packet.data)
 	var isAckEliciting bool
+	var recoverableFrame wire.Frame
+
+	if s.mitigationOn(packet.encryptionLevel) {
+		var ignorableError *qerr.QuicError
+		var parsedFirstFrame bool
+		validatingReader := bytes.NewReader(packet.data)
+		for {
+			frame, err := s.frameParser.ParseNext(validatingReader, packet.encryptionLevel)
+			if err != nil {
+				quicErr, ok := err.(*qerr.QuicError)
```

Comment:
Nit: can be combined with the `if` statement.

Reply:
done